cpp/arcticdb/version/version_map_entry.hpp

```diff
@@ -31,6 +31,17 @@ enum class LoadType :
     LOAD_ALL = 6
 };
 
+enum class VersionStatus {
+    LIVE,
+    TOMBSTONED,
+    ABSENT_IN_STORAGE
```

Comment:
How about `NEVER_EXISTED`? Absent in storage sounds like it could exist but we failed to read it or something, but that isn't the logical state we're trying to represent here.